src/hotspot/share/nmt/memflags.cpp

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2012, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#include "precompiled.hpp"
+
+#include "nmt/memflags.hpp"
+#include "utilities/debug.hpp"
+
+// Extra insurance that MEMFLAGS truly has the same size as uint8_t.
+STATIC_ASSERT(sizeof(MEMFLAGS) == sizeof(uint8_t));
```

Comment:
I think you can remove this entire .cpp file. There's no need to check the size of an enum with a specified base type.

Reply:
I rather have this explicit check. If MEMFLAGS>1byte, things break, and I would like to make that explicit.

That said, I can move this static assert to the header. I just wanted to avoid including debug.hpp. Note that the cpp file would be the right place in the future for any MEMFLAGS related utility functions, e.g. to-and-from-string conversations.

Reply:
Could you instead put the static_assert near the code that will break? Right now it looks obscure and weird to have this check when it is obviously correct as long as no one changes the definition. Would it be enough to write a comment in the header that this needs to be 1 byte?

Reply:
To quote @robehn - Why write a comment for a rule if you can enforce it with code instead...

Reply:
I tend to agree with that. My earlier question still stands is there a better place to put it? Right now the "enforced" code in a stand-alone file doesn't tell me "why" this is important.

Reply:
If you want to keep the static_assert it in the .cpp file, then I won't block that.

Reply:
> Could you instead put the static_assert near the code that will break?
I like that. I will do that.